src/util/StringUtils.cpp

```diff
@@ -39,6 +39,23 @@ string getUppercase(const string& orig) {
   return retVal;
 }
 
+// ____________________________________________________________________________
+bool strIsLangTag(const string& input) {
+  auto lenInput = input.length();
+  if (lenInput < 2) {
+    return false;
+  }
+  UErrorCode status = U_ZERO_ERROR;
+  char localeID[ULOC_FULLNAME_CAPACITY];
+  int32_t parsedLength = 0;
+  uloc_forLanguageTag(input.c_str(), localeID, ULOC_FULLNAME_CAPACITY,
+                      &parsedLength, &status);
+  if (U_FAILURE(status) || parsedLength != static_cast<int32_t>(lenInput)) {
+    return false;
+  }
+  return true;
```

Comment:
I think this is overkill 
and costs us time for not that much gain.
A much simpler implementation would be to check against the same Regex that the SPARQL grammar wants:
 `[a-zA-Z]+ ('-' [a-zA-Z0-9]+)*`
So something like

`return ctre::match<"[a-zA-Z]+(-[a-zA-Z0-9]+)*">`

should do the trick (verify in unit tests that this regex does the right thing).